sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -247,10 +274,19 @@ Scheduler::GraphBuilder::findDepsForReq(MemObjRecord *Record, Requirement *Req,
                                         QueueImplPtr Queue) {
   sycl::context Context = Queue->get_context();
   std::set<Command *> RetDeps;
-  std::vector<Command *> ToAnalyze = Record->MLeafs;
   std::set<Command *> Visited;
   const bool ReadOnlyReq = Req->MAccessMode == access::mode::read;
 
+  std::vector<Command *> ToAnalyze;
+
+  if (ReadOnlyReq) {
```

Comment:
If you remove extra bracers would you be given a dangling-else warning? If no, please remove them.